### DISCUSSION TITLE
Proxmox 8.1 and Truenas Core Problem

### DISCUSSION THREAD

**Discussion by cundaric** [TheGrandWazoo/freenas-proxmox#181] [Q&A] [answered]
ello,
I followed step by step the installation instructions but I'm getting the following error when trying to execute qm rescan command on Proxmox 8.1 and Truesan Core
Unable to connect to the FreeNAS API service at '192.168.20.7' using the 'https' protocol at /usr/share/perl5/PVE/Storage/LunCmd/FreeNAS.pm line 380.
Plugin version
pkg -l |grep freenas
ii freenas-proxmox 2.2.0-1 all FreeNAS Plugin for Proxmox VE
line 380 "die "Unable to connect to the FreeNAS API service at '" . $apihost . "' using the '" . $scheme . "' protocol";"

IP is reachable on the port:
curl -k -I https://192.168.20.7/api/
HTTP/2 302
server: nginx
date: Thu, 21 Mar 2024 08:57:51 GMT
content-type: text/html
content-length: 138
location: https://192.168.20.7/ui/
strict-transport-security: max-age=63072000; includeSubDomains; preload
x-content-type-options: nosniff
x-xss-protection: 1; mode=block
permissions-policy: geolocation=(),midi=(),sync-xhr=(),microphone=(),camera=(),magnetometer=(),gyroscope=(),fullscreen=(self),payment=()
referrer-policy: strict-origin
x-frame-options: SAMEORIGIN

Any ideass?

**Comment by TheGrandWazoo** [comment, marked as answer]
Sorry for the long silence on this thread. This is the old `LunCmd/FreeNAS.pm` API driver from v2.2.0 — a lot has changed since then. v3.0 is a full rewrite that talks to the TrueNAS v2.0 REST API directly (bearer token auth only, no SSH, no basic auth) and has been confirmed working on TrueNAS CORE 13.0-U6. If you're still hitting connection issues on CORE, I'd recommend upgrading to v3.0 rather than debugging the old driver — see the [README](README.md) for setup and [Migrating from v2.x](docs/migrating-from-v2.md) if you have existing v2.x storage to carry over.
